contracts/modules/tokenswap/TokenSwapModule.sol

```diff
@@ -238,33 +239,36 @@ contract TokenSwapModule is ModuleBaseWithFee {
                 "TokenSwapModule: Error 102"
             );
         }
-        // ToDo: Add comment
         uint256 daoplomatLen = _daoplomats.length;
-        if (daoplomatLen == 0) {
+        // If no DAOplomat reward is set
+        if (daoplomatLen == 0 || _rewardPathTo[0][0] == 0) {
             require(
-                daoplomatLen == _rewardPathTo[0].length &&
+                daoplomatLen == _rewardPathTo[0][0] &&
                     daoplomatLen == _rewardPathTo[1].length,
                 "TokenSwapModule: Error 102"
             );
-            // } else {}
+        } else {
+            // Max number of DAOplomats
             require(daoplomatLen <= 8, "TokenSwapModule: Error 267");
```

Comment:
Do we want an immutable public variable for this? Looks a bit opaque ("why is that 8, where does that come from?"). What do you think?

Reply:
Agree